src/util.ts

```diff
@@ -85,17 +85,16 @@ export function isGhes(): boolean {
 }
 
 export function isCacheFeatureAvailable(): boolean {
-  if (!cache.isFeatureAvailable()) {
-    if (isGhes()) {
-      throw new Error(
-        'Caching is only supported on GHES version >= 3.5. If you are on a version >= 3.5, please check with your GHES admin if the Actions cache service is enabled or not.'
-      );
-    } else {
-      core.warning('The runner was not able to contact the cache service. Caching will be skipped');
-    }
+  if (cache.isFeatureAvailable()) {
+    return true;
+  }
 
-    return false;
+  if (isGhes()) {
+    throw new Error(
+      'Caching is only supported on GHES version >= 3.5. If you are on a version >= 3.5, please check with your GHES admin if the Actions cache service is enabled or not.'
+    );
```

Comment:
How about replace throw with log the warning here and return `false`?

Reply:
Yes, let's apply the same updates we did on setup-go as well :)

Reply:
@panticmilos Thank you for your review. I pushed a commit.